### ISSUE TITLE
Reject all or none of the commits of a complex operation

### ISSUE THREAD

**Issue by bdionne** [NCIEVS/nci-protege5#296]
A complex operation such as merge or retire, will involve several commits.

Prevent the user for rejecting the commits piecemeal. If the user selects a commit that is part of a complex operation, the reject button will not be enabled until all the others are selected.

In other words the entire operation is rolled back.

**Comment by rmehri** [comment]
Hi,
This fix has been retested verified using nci-1.0068 and edit-tab-0.6.3.
1) Retested and verified for pre-tired and pre-merged classes. Used Manager login -  Under Revision History tab, all tabs greyed out, (Passed)
2) Retested and verified for retired and merged classes. Used Manager login -  Under Revision History tab, all tabs greyed out, except the Squash and history tabs are enabled. Passed
Please see attached snapshots for more details.

![issues-296-preretired](https://user-images.githubusercontent.com/8367102/27143411-44d1d830-50fc-11e7-8863-37e0ce2b3a10.png)
![merge by manager_issue 2
![issue-296-retired](https://user-images.githubusercontent.com/8367102/27143504-91848290-50fc-11e7-84aa-62b64ece8538.png)
96-1](https://user-images.githubusercontent.com/8367102/27143450-662248bc-50fc-11e7-92c5-381922734d9d.png)
![premerged](https://user-images.githubusercontent.com/8367102/27143978-352884b8-50fe-11e7-8866-96f747a7931c.png)

Thanks,
Reza

**Comment by rmehri** [comment]
Hi,
 This fix has been retested verified using nci-1.0069 and edit-tab-0.6.5.
1.Retested and verified for pre-tired and pre-merged classes. Used Manager login - Under Revision History tab, all tabs greyed out, (Passed)
2.Retested and verified for retired and merged classes. Used Manager login - Under Revision History tab, all tabs greyed out, except the Squash and history tabs are enabled. ( Passed_
 Please see attached snapshots for more details.

![merged](https://user-images.githubusercontent.com/8367102/27295947-d6ff7e58-54ec-11e7-8158-d7c9ff88b278.png)
![pre-merged](https://user-images.githubusercontent.com/8367102/27295949-d70081cc-54ec-11e7-85fb-b1769a8f9700.png)
![pre retired](https://user-images.githubusercontent.com/8367102/27295946-d6fe8d22-54ec-11e7-83a3-34f28069b28c.png)
![retired](https://user-images.githubusercontent.com/8367102/27295948-d700849c-54ec-11e7-8459-e041cb4cb403.png)

Thanks,
Reza

**Comment by bdionne** [comment]
@rmehri - I'm curious, did you test the main issue in the subject line? These scenarios seem related to whether or not a user can use the `squash` button. I don't see anything about the rejections?

**Comment by rmehri** [comment]
Hi Bob, 
I am not clear. Under ComplexEdit/ Complex operation page, we have Split, Copy, Merge and Dual Edits.   
Does this fix include all complex operation? Or, a few? Please clarify so I can retest and verify them again.
Thanks,
Reza  

![complex operations](https://user-images.githubusercontent.com/8367102/27340783-5df06898-55a9-11e7-8994-c161b5a2298c.png)

Bob Stated:
A complex operation such as merge or retire, will involve several commits.
Prevent the user for rejecting the commits piecemeal. If the user selects a commit that is part of a complex operation, the reject button will not be enabled until all the others are selected.
In other words the entire operation is rolled back.

**Comment by bdionne** [comment]
@rmehri - the feature involves the `rejecting` of the commits of a complex operation. This happens in the `Revision Diff` panel. A complex operation will usually involve multiple changes. So .eg. here's what a retire looks like:

<img width="1090" alt="screen shot 2017-06-20 at 2 29 19 pm" src="https://user-images.githubusercontent.com/31741/27349468-59bbd66a-55c5-11e7-8ee2-6bb8e8ebbdf8.png">

Notice the reject button is disabled if you select any of those changes. However if you select all of them, you'll see the `reject` button become enabled. This forces the user to reject all of them or none of them:

<img width="1088" alt="screen shot 2017-06-20 at 2 29 36 pm" src="https://user-images.githubusercontent.com/31741/27349526-8b90fd46-55c5-11e7-97b1-555013b87347.png">

**Comment by rmehri** [comment]
Hi Bob,
Thanks for you feedback, this has been retested good now and yesterday. Please see attached screenshots for more detail.
![merged](https://user-images.githubusercontent.com/8367102/27356133-66b5485e-55db-11e7-8f16-0451d529d8d7.png)
![merged1](https://user-images.githubusercontent.com/8367102/27356141-6df81a74-55db-11e7-8407-ea5bfce35088.png)

Thanks,
Reza
